specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2024-04-01/mfe.json

```diff
@@ -19466,11 +19813,374 @@
+          "description": "[Required] Input path type for package inputs.",
+          "$ref": "#/definitions/InputPathType"
+        }
+      },
+      "discriminator": "inputPathType"
+    },
+    "PackageInputPathId": {
+      "description": "Package input path specified with a resource id.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/PackageInputPathBase"
+        }
+      ],
+      "properties": {
+        "resourceId": {
+          "description": "Input resource id.",
+          "type": "string",
+          "x-nullable": true
+        }
+      },
+      "x-ms-discriminator-value": "PathId",
+      "additionalProperties": false
+    },
+    "PackageInputPathUrl": {
+      "description": "Package input path specified as an url.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/PackageInputPathBase"
+        }
+      ],
+      "properties": {
+        "url": {
+          "description": "Input path url.",
+          "type": "string",
+          "x-nullable": true
+        }
+      },
+      "x-ms-discriminator-value": "Url",
+      "additionalProperties": false
+    },
+    "PackageInputPathVersion": {
+      "description": "Package input path specified with name and version.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/PackageInputPathBase"
+        }
+      ],
+      "properties": {
+        "resourceName": {
+          "description": "Input resource name.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "resourceVersion": {
+          "description": "Input resource version.",
+          "type": "string",
+          "x-nullable": true
+        }
+      },
+      "x-ms-discriminator-value": "PathVersion",
+      "additionalProperties": false
+    },
+    "PackageInputType": {
+      "description": "Type of the inputs.",
+      "enum": [
+        "UriFile",
+        "UriFolder"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "PackageInputType",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "UriFile"
+          },
+          {
+            "value": "UriFolder"
+          }
+        ]
+      },
+      "additionalProperties": false
+    },
+    "PackageRequest": {
+      "description": "Model package operation request properties.",
+      "required": [
+        "inferencingServer",
+        "targetEnvironmentId"
+      ],
+      "type": "object",
+      "properties": {
+        "baseEnvironmentSource": {
+          "description": "Base environment to start with.",
+          "$ref": "#/definitions/BaseEnvironmentSource",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ],
+          "x-nullable": true
+        },
+        "environmentVariables": {
+          "description": "Collection of environment variables.",
+          "type": "object",
+          "additionalProperties": {
```

Comment:
>additionalProperties [](http://example.com/codeflow?start=11&length=20)

This is an ARM anti-pattern and generally not allowed. Can you model this as an array of (string, string) tuples?